angular-chess/src/app/services/move-execution.service.ts

```diff
@@ -74,6 +74,10 @@ export class MoveExecutionService {
   }
 
   private finishMove(move: Move): void {
+    if (!(move.piece.type === PieceType.PAWN && Math.abs(move.from.row - move.to.row) === 2)) {
```

Comment:
2 = magic number - please create a const

Reply:
How many steps can go a PAWN? 1 or 2, correct?
Alternative suggestion for a better readability: to extract the condition as extra method: hasPawnGoneLongStep

Reply:
I added the suggested method. Normally a pawn can go only one step. It can go two steps on it's initial row. Please see https://www.fide.com/FIDE/handbook/LawsOfChess.pdf for the official rules.